tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Execution/ProcessManager.cs

```diff
@@ -345,5 +365,31 @@ static List<int> GetChildrenPS (ILog log, int pid)
 
 			return list;
 		}
+
+		static string DetectXcodePath ()
+		{
+			using var process = new Process ();
+			process.StartInfo.FileName = "xcode-select";
+			process.StartInfo.Arguments = "-p";
+
+			var log = new MemoryLog ();
+			var stdout = new MemoryLog () { Timestamp = false };
+			var stderr = new ConsoleLog ();
+			var timeout = TimeSpan.FromSeconds (30);
+
+			var result = RunAsyncInternal (process, log, stdout, stderr, timeout).GetAwaiter ().GetResult ();
+
+			if (!result.Succeeded)
+				throw new Exception ("Failed to detect Xcode path from xcode-select!");
```

Comment:
You have two different failure conditions that throw identical exceptions. Future you will be happier if you can distinguish the two cases.

Reply:
Good point. Too bad it got merged already. However if this fails, we will have details in stderr (which is a ConsoleLog) so I guess we will be able to debug